src/components/topNav/primaryTopNav/PrimaryTopNav.tsx

```diff
@@ -35,6 +36,20 @@ export const PrimaryTopNav = ({ items, mobileLogo, layoutClassName }: TopNavProp
     }
   };
 
+  const handleSubItemMenu = (idx: number, value: boolean) => {
+    setSubNavIsOpen({
+      ...subNavIsOpen,
+      [idx as keyof typeof subNavIsOpen]: value,
+    });
+  };
```

Comment:
This seems like a complex way to keep track of the open sub items. Previously, I think, we just let the css handle everything. Why the change?